gef.py

```diff
@@ -8658,81 +8658,97 @@ def get_settings_path(self):
         return path if os.path.isdir(path) else None
 
 
-class GenericOffsetFunction(gdb.Function):
-    """This is an abstract class for invoking offset functions, should not be instantiated."""
+@lru_cache()
+def get_section_base_address(name):
+    section = process_lookup_path(name)
+    if section:
+        return section.page_start
+
+    return None
+
+@lru_cache()
+def get_zone_base_address(name):
+    zone = file_lookup_name_path(name, get_filepath())
+    if zone:
+        return zone.zone_start
+
+    return None
+
+class GenericFunction(gdb.Function):
+    """This is an abstract class for invoking convenience functions, should not be instantiated."""
     __metaclass__ = abc.ABCMeta
 
-    @abc.abstractproperty
-    def _section_(self): pass
-    @abc.abstractproperty
-    def _zone_(self): pass
     @abc.abstractproperty
     def _function_(self): pass
     @property
     def _syntax_(self):
         return "${}([offset])".format(self._function_)
 
     def __init__ (self):
-        super(GenericOffsetFunction, self).__init__(self._function_)
+        super(GenericFunction, self).__init__(self._function_)
 
-    def invoke(self, offset=gdb.Value(0)):
+    def invoke(self, *args):
         if not is_alive():
             raise gdb.GdbError("No debugging session active")
+        return long(self.do_invoke(args))
+
+    def arg_to_long(self, args, index, default=0):
+        try:
+            addr = args[index]
```

Comment:
1. Why pass in `args` and `index` and do `args[index]` when you can just have a more general `value_to_long`? is it to help with catching index errors in the caller? That makes sense, but I think naming could be more clear.

Reply:
Yeah everything that implements this class can either have no arguments and the offset defaults to 0 or one argument that is the offset. So there was just a lot of code duplication. I could add it as value_to_long and have another `GenericOffsetFunction` class that handles setting the default? Then it's not tied to `GenericFunction`

Reply:
Naw it's fine as it is, I had to check to make sure that the args at `gdb.Value`s. Makes sense to keep a helper like this in the base class, I am sure more than just offset functions would want something like it.